src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsDao.java

```diff
@@ -44,13 +47,26 @@ public SnapshotBuilderSettings getSnapshotBuilderSettingsByDatasetId(UUID datase
 
       return jdbcTemplate.queryForObject(
           "SELECT settings FROM snapshot_builder_settings WHERE dataset_id = :dataset_id",
-          Map.of(datasetIdField, datasetId),
+          Map.of(DATASET_ID_FIELD, datasetId),
           MAPPER);
     } catch (EmptyResultDataAccessException ex) {
       throw new NotFoundException("No snapshot builder settings found for dataset", ex);
     }
   }
 
+  @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+  public SnapshotBuilderSettings getSnapshotBuilderSettingsBySnapshotId(UUID snapshotId) {
```

Comment:
I may have missed this the first time around, but I would use a shorter name for this method. This method is in a class called `SnapshotBuilderSettingsDao` and will only be called via an object with the same name. So it seems redundant to put `snapshotBuilderSettings` in the method name too.
```suggestion
  public SnapshotBuilderSettings getBySnapshotId(UUID snapshotId) {
```